File: 389.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        # time: O(n), space: O(n)
        cnt = {}
        for c in s:
            cnt[c] = cnt.get(c, 0) + 1
        for c in t:
            if c not in cnt or cnt[c] == 0:
                return c
            cnt[c] -= 1

File: 389.java
class Solution {
    public char findTheDifference(String s, String t) {
        // time: O(m + n), space: O(1)
        char c = 0;
        for (char cs : s.toCharArray()) c ^= cs;
        for (char ct : t.toCharArray()) c ^= ct;
        return c;
    }
}